[CI-4786] Add removing search term on track recommendation click

Alexey-Pavlov opened the pull request:
This pull request adds a new utility function for tracking recommendation selections and updates the `useDownShift` hook to use this new function. The main idea is that we have removing `SEARCH_TERM_STORAGE_KEY` in `src/tracker.js` in [autocomplete-ui](https://github.com/Constructor-io/autocomplete-ui/blob/master/src/tracker.js#L437C1-L438C1), but in this library, this part is missing. I added it here, so customers who use our autocomplete library don't face issues when we send the old search term by clicking on recommendations and adding to cart. 

Also, this PR improves how recommendation clicks are tracked, making the tracking logic more modular and reusable.

**Tracking improvements:**

* Added a new `trackRecommendationSelect` function to `src/utils/tracking.ts` to encapsulate recommendation click tracking logic and ensure consistent storage cleanup.
* Updated the `useDownShift` hook in `src/hooks/useDownShift.ts` to use the new `trackRecommendationSelect` function instead of calling the tracker directly, improving maintainability and separation of concerns.
* Added `storageRemoveItem` to remove the old `SEARCH_TERM_STORAGE_KEY`

mocca102 reviewed (commented):
Looking great!

Can we add a test case similar to this in `ComponentTests.stories.tsx`
` expect(storageGetItem(CONSTANTS.SEARCH_TERM_STORAGE_KEY)).toBeNull();`

The tests seem to be failing as well (I don't think it's related to your PR) but we need to fix it

Alexey-Pavlov commented:
> Looking great!
> 
> Can we add a test case similar to this in `ComponentTests.stories.tsx` ` expect(storageGetItem(CONSTANTS.SEARCH_TERM_STORAGE_KEY)).toBeNull();`
> 
> The tests seem to be failing as well (I don't think it's related to your PR), but we need to fix it

@mocca102 Thank you for the review! It was interesting 🧠
I added: 

- [test](https://github.com/Constructor-io/constructorio-ui-autocomplete/pull/230/commits/ad7784da61fc9c89a4683c32c72bb4dafca02e68#diff-9e672232b6daaac609339711f7f0bc7768b65ec952b71e5e2a733a50f38866e1R390), where I add something to the store using storageSetItem, then open zero state with recs, click on one of them, and then check that SEARCH_TERM_STORAGE_KEY is null 
- [fixed](https://github.com/Constructor-io/constructorio-ui-autocomplete/pull/230/commits/ad7784da61fc9c89a4683c32c72bb4dafca02e68#diff-9e672232b6daaac609339711f7f0bc7768b65ec952b71e5e2a733a50f38866e1R557) few getAllByText, because actually text there looks like `<p class="cio-term-in-group">in Socks &amp; Underwear</p>`
- FocusFiresTrackingEvent tests were flaky, sometimes passed, sometimes not, so I overwrote `Storage.prototype.setItem`, I don't like it, but this way it works stably without surprises (sometimes `_constructorio_requests` appears to be deleted at the moment of check), and extracted it to its own helper to reduce code duplication

mocca102 commented:
> * FocusFiresTrackingEvent tests were flaky, sometimes passed, sometimes not, so I overwrote `Storage.prototype.setItem`, I don't like it, but this way it works stably without surprises (sometimes `_constructorio_requests` appears to be deleted at the moment of check), and extracted it to its own helper to reduce code duplication

Actually, I like this approach better. This LGTM.

I'll approve it but please make sure to get an approval from a CDX team member as well

Alexey-Pavlov commented on src/stories/tests/ComponentTests.stories.tsx:
```diff
@@ -357,6 +362,50 @@ SelectProductSuggestionClearsSearchTermStorage.play = async ({ canvasElement })
+    await userEvent.click(firstRecommendation);
+  }
+
+  expect(storageGetItem(CONSTANTS.SEARCH_TERM_STORAGE_KEY)).toBeNull();
```
Yep, we can, without any problems 
Added this check ✅
